README.md
<p align="center">
  <img src="https://raw.githubusercontent.com/luyadev/luya/master/docs/logo/luya-logo-0.2x.png" alt="LUYA Logo"/>
</p>

# Bootstrap 3

[![LUYA](https://img.shields.io/badge/Powered%20by-LUYA-brightgreen.svg)](https://luya.io)
[![Build Status](https://travis-ci.org/luyadev/luya-bootstrap3.svg?branch=master)](https://travis-ci.org/luyadev/luya-bootstrap3)
[![Test Coverage](https://api.codeclimate.com/v1/badges/c4debece5963e70fb2ff/test_coverage)](https://codeclimate.com/github/luyadev/luya-bootstrap3/test_coverage)
[![Latest Stable Version](https://poser.pugx.org/luyadev/luya-bootstrap3/v/stable)](https://packagist.org/packages/luyadev/luya-bootstrap3)
[![Total Downloads](https://poser.pugx.org/luyadev/luya-bootstrap3/downloads)](https://packagist.org/packages/luyadev/luya-bootstrap3)
[![Slack Support](https://img.shields.io/badge/Slack-luyadev-yellowgreen.svg)](https://slack.luya.io/)

Bootstrap 3 related content:

+ blocks

## Installation

For the installation of modules Composer is required.

```sh
composer require luyadev/luya-bootstrap3
```


CHANGELOG.md
# CHANGELOG

All notable changes to this project will be documented in this file. This project adheres to [Semantic Versioning](http://semver.org/).
In order to read more about upgrading and BC breaks have a look at the [UPGRADE Document](UPGRADE.md).

## 1.0.11 (26. July 2023)

+ Fixed php 8 issues.

## 1.0.10 (9. December 2021)

+ Small changes in docs, translations, composer dependencies

## 1.0.9 (25. March 2021)

+ [#31](https://github.com/luyadev/luya-bootstrap3/pull/31) Added Bulgarian translations.

## 1.0.8 (25. February 2020)

+ New PT translations

## 1.0.7 (22. October 2019)

+ New FR translations

## 1.0.6 (14. May 2019)

+ [#16](https://github.com/luyadev/luya-bootstrap3/issues/16) Added google maps api key option.
+ [#26](https://github.com/luyadev/luya-bootstrap3/pull/26) Added polish translations.
+ Use new {{DS}} separator for block paths.

## 1.0.5 (3. December 2018)

+ [#23](https://github.com/luyadev/luya-bootstrap3/issues/23) Fixed bug with ImageBlock not linking to hidden menu items.

## 1.0.4 (4. September 2018)

+ [#17](https://github.com/luyadev/luya-bootstrap3/issues/17) Updated to bootstrap 3 media text for image text block.
+ [#11](https://github.com/luyadev/luya-bootstrap3/issues/11) Turkish added
+ [#15](https://github.com/luyadev/luya-bootstrap3/issues/18) Dutch added
+ [#9](https://github.com/luyadev/luya-bootstrap3/issues/9) French update

## 1.0.3 (1. June 2018)

+ [#8](https://github.com/luyadev/luya-bootstrap3/issues/8) Fixed missing translation of image block labels.

## 1.0.2 (7. May 2018)

+ [#6](https://github.com/luyadev/luya-bootstrap3/pull/6) Embed maps use the composition localisation lang short code.
+ [#5](https://github.com/luyadev/luya-bootstrap3/issues/5) Added Snazzy Maps iFrame URL field to MapBlock and updated translations.
+ [#4](https://github.com/luyadev/luya-bootstrap3/pull/4) Translated FormBlock from German to English.

## 1.0.1 (17. January 2018)

+ [#3](https://github.com/luyadev/luya-bootstrap3/issues/3) Add alt and title from image caption.

## 1.0.0 (12. December 2017)

+ First stable release.


src/blocks/MapBlock.php
<?php

namespace luya\bootstrap3\blocks;

use Yii;
use luya\bootstrap3\Module;
use luya\bootstrap3\BaseBootstrap3Block;
use luya\bootstrap3\blockgroups\Bootstrap3Group;

/**
 * Google Maps Block.
 *
 * @author Basil Suter <basil@nadar.io>
 * @since 1.0.0
 */
final class MapBlock extends BaseBootstrap3Block
{
    /**
     * @inheritdoc
     */
    public $cacheEnabled = true;

    /**
     * @inheritdoc
     */
    public function name()
    {
        return Module::t('block_map_name');
    }
    
    public function blockGroup()
    {
        return Bootstrap3Group::class;
    }

    /**
     * @inheritdoc
     */
    public function icon()
    {
        return 'map';
    }

    /**
     * @inheritdoc
     */
    public function config()
    {
        return [
            'vars' => [
                [
                    'var' => 'address',
                    'label' => Module::t('block_map_address_label'),
                    'type' => self::TYPE_TEXT,
                    'placeholder' => 'Zephir Software Design AG, Tramstrasse 66, 4142 Münchenstein'
                ],
                [
                    'var' => 'zoom',
                    'label' => Module::t('block_map_zoom_label'),
                    'type' => self::TYPE_SELECT,
                    'initvalue' => '12',
                    'options' => [
                        ['value' => '0', 'label' => Module::t('block_map_zoom_entire')],
                        ['value' => '1', 'label' => '4000 km'],
                        ['value' => '2', 'label' => '2000 km (' . Module::t('block_map_zoom_world') . ')'],
                        ['value' => '3', 'label' => '1000 km'],
                        ['value' => '4', 'label' => '400 km (' . Module::t('block_map_zoom_continent') . ')'],
                        ['value' => '5', 'label' => '200 km'],
                        ['value' => '6', 'label' => '100 km (' . Module::t('block_map_zoom_country') . ')'],
                        ['value' => '7', 'label' => '50 km'],
                        ['value' => '8', 'label' => '30 km'],
                        ['value' => '9', 'label' => '15 km'],
                        ['value' => '10', 'label' => '8 km'],
                        ['value' => '11', 'label' => '4 km'],
                        ['value' => '12', 'label' => '2 km (' . Module::t('block_map_zoom_city') . ')'],
                        ['value' => '13', 'label' => '1 km'],
                        ['value' => '14', 'label' => '400 m (' . Module::t('block_map_zoom_district') . ')'],
                        ['value' => '15', 'label' => '200 m'],
                        ['value' => '16', 'label' => '100 m'],
                        ['value' => '17', 'label' => '50 m (' . Module::t('block_map_zoom_street') . ')'],
                        ['value' => '18', 'label' => '20 m'],
                        ['value' => '19', 'label' => '10 m'],
                        ['value' => '20', 'label' => '5 m (' . Module::t('block_map_zoom_house') . ')'],
                        ['value' => '21', 'label' => '2.5 m'],
                    ],
                ],
                [
                    'var' => 'maptype',
                    'label' => Module::t('block_map_maptype_label'),
                    'type' => self::TYPE_SELECT,
                    'options' => [
                        ['value' => 'm', 'label' => Module::t('block_map_maptype_roadmap')],
                        ['value' => 'k', 'label' => Module::t('block_map_maptype_satellitemap')],
                        ['value' => 'h', 'label' => Module::t('block_map_maptype_hybrid')],
                    ],
                ],
            ],
            'cfgs' => [
                ['var' => 'snazzymapsUrl', 'label' => Module::t('block_map_snazzymapsUrl_label'), 'type' => self::TYPE_TEXT],
                ['var' => 'mapsKey', 'label' => Module::t('block_map_api_key'), 'type' => self::TYPE_TEXT],
            ]
        ];
    }

    public function getFieldHelp()
    {
        return [
            'snazzymapsUrl' => Module::t('block_map_snazzymapsUrl_help'),
            'mapsKey' => Module::t('block_map_api_key_help'),
        ];
    }

    /**
     * @inheritdoc
     */
    public function extraVars()
    {
        return [
            'embedUrl' => $this->generateEmbedUrl(),
        ];
    }

    /**
     * Generate the embed url based on localisation or whether its snazzy or not.
     * @since 1.0.2
     */
    public function generateEmbedUrl()
    {
        if (($snazzymapsUrl = $this->getCfgValue('snazzymapsUrl'))) {
            return $snazzymapsUrl;
        }
        
        if (empty($this->getVarValue('address'))) {
            return false;
        }
        
        $params = [
            'f' => 'q',
            'source' => 's_q',
            'hl' => Yii::$app->composition->langShortCode,
            'q' => $this->getVarValue('address', 'Zephir Software Design AG, Tramstrasse 66, Münchenstein'),
            'z' => $this->getVarValue('zoom', 15),
            't' => $this->getVarValue('maptype', 'h'),
            'output' => 'embed',
            'key' => $this->getCfgValue('mapsKey'),
        ];
        
        return 'https://maps.google.com/maps?' . http_build_query(array_filter($params), '', '&', PHP_QUERY_RFC1738);
    }

    /**
     * @inheritdoc
     */
    public function admin()
    {
        return '{% if vars.address is not empty %}<div class="iframe-container"><iframe src="{{extras.embedUrl | raw}}"></iframe></div>{% else %}<span class="block__empty-text">' . Module::t('block_map_no_content') . '</span>{% endif %}';
    }
}
